MvvmHelpers/WeakEventManager.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+using static System.String;
+
+namespace MvvmHelpers
+{
+	public class WeakEventManager
+	{
+		readonly Dictionary<string, List<Subscription>> eventHandlers = new Dictionary<string, List<Subscription>>();
+
+		public void AddEventHandler<TEventArgs>(EventHandler<TEventArgs> handler, [CallerMemberName]string eventName = null)
+			where TEventArgs : EventArgs
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
```

Comment:
Use `handler is null` to squeeze out some more performance benefits: https://www.gullberg.tk/blog/is-null-versus-null-in-c/

**tl;dr** Pattern Matching is faster than Equality comparison in the .NET IL